dipy/workflows/reconst.py

```diff
@@ -0,0 +1,151 @@
+        out_evec : string, optional
+            Name of the eigen vectors volume to be saved
+            (default 'evecs.nii.gz')
+        out_eval : string, optional
+            Name of the eigen vvalues to be saved (default 'evals.nii.gz')
+        """
+
+        io_it = self.get_io_iterator()
+
+        for dwi, bval, bvec, mask, otensor, ofa, oga, orgb, omd, oad, ord, \
+            omode, oevecs, oevals in io_it:
+
+            logging.info('Computing dti metrics for {0}'.format(dwi))
+            img = nib.load(dwi)
+            data = img.get_data()
+            affine = img.get_affine()
+
+            if mask is None:
+                mask = None
+            else:
+                mask = nib.load(mask).get_data().astype(np.bool)
+
+            tenfit, _ = get_fitted_tensor(data, mask, bval, bvec, b0_threshold)
+
+            FA = fractional_anisotropy(tenfit.evals)
+            FA[np.isnan(FA)] = 0
+            FA = np.clip(FA, 0, 1)
+
+            tensor_vals = lower_triangular(tenfit.quadratic_form)
+            correct_order = [0, 1, 3, 2, 4, 5]
+            tensor_vals_reordered = tensor_vals[..., correct_order]
+            fiber_tensors = nib.Nifti1Image(tensor_vals_reordered.astype(
+                np.float32), affine)
+            nib.save(fiber_tensors, otensor)
+
+            fa_img = nib.Nifti1Image(FA.astype(np.float32), affine)
+            nib.save(fa_img, ofa)
+
+            GA = geodesic_anisotropy(tenfit.evals)
+            ga_img = nib.Nifti1Image(GA.astype(np.float32), affine)
+            nib.save(ga_img, oga)
+
+            RGB = color_fa(FA, tenfit.evecs)
+            rgb_img = nib.Nifti1Image(np.array(255 * RGB, 'uint8'), affine)
+            nib.save(rgb_img, orgb)
+
+            MD = mean_diffusivity(tenfit.evals)
+            md_img = nib.Nifti1Image(MD.astype(np.float32), affine)
+            nib.save(md_img, omd)
+
+            AD = axial_diffusivity(tenfit.evals)
+            ad_img = nib.Nifti1Image(AD.astype(np.float32), affine)
+            nib.save(ad_img, oad)
+
+            RD = radial_diffusivity(tenfit.evals)
+            rd_img = nib.Nifti1Image(RD.astype(np.float32), affine)
+            nib.save(rd_img, ord)
+
+            MODE = get_mode(tenfit.quadratic_form)
+            mode_img = nib.Nifti1Image(MODE.astype(np.float32), affine)
+            nib.save(mode_img, omode)
+
+            evecs_img = nib.Nifti1Image(tenfit.evecs.astype(np.float32), affine)
+            nib.save(evecs_img, oevecs)
+
+            evals_img = nib.Nifti1Image(tenfit.evals.astype(np.float32), affine)
+            nib.save(evals_img, oevals)
+            logging.info('All dti metrics saved in {0}'.
+                         format(os.path.dirname(oevals)))
+
+
+def get_fitted_tensor(data, mask, bval, bvec, b0_threshold=0):
+    logging.info('Tensor estimation...')
+    bvals, bvecs = read_bvals_bvecs(bval, bvec)
+    gtab = gradient_table(bvals, bvecs, b0_threshold=b0_threshold)
+
+    tenmodel = TensorModel(gtab)
```

Comment:
From what I can see there is no way to do it that is not very ugly and error prone. If you guys have an idea im really open to it.

Right now what I can see working is adding a string param and parsing it as a dict for kwargs but im not sure it is the kind of solution you were thinking about. 

Also, I might make generic dti_reconst workflow and make a child for each tensor model (with good params).  This would actually make a callabable script per tensor_model, probablu not what you wanted too. 

Or like I said just add the three params possibilities and use it when needed.